azurelinuxagent/common/protocol/wire.py

```diff
@@ -623,103 +596,141 @@ def call_storage_service(http_req, *args, **kwargs):
 
         return http_req(*args, **kwargs)
 
-    def fetch_manifest_through_host(self, uri):
-        host = self.get_host_plugin()
-        uri, headers = host.get_artifact_request(uri)
-        response, _ = self.fetch(uri, headers, use_proxy=False, retry_codes=restutil.HGAP_GET_EXTENSION_ARTIFACT_RETRY_CODES)
-        return response
+    def fetch_artifacts_profile_blob(self, uri):
```

Comment:
This is where the refactoring of the download logic begins. Most of the logic is in _download_with_fallback_channel() and _download_using_appropriate_channel(). fetch_artifacts_profile_blob(), fetch_manifest() and download_extension() just define the functions to use for the direct and HGAP call and then invoke _download_with_fallback_channel()